pkg/hooks/manager.go

```diff
@@ -207,6 +216,52 @@ func (m *Manager) OnDefineDomain(domainSpec *virtwrapApi.DomainSpec, vmi *v1.Vir
 	return string(domainSpecXML), nil
 }
 
+func (m *Manager) PreCloudInitIso(vmi *v1.VirtualMachineInstance, source *v1.CloudInitNoCloudSource) (*v1.CloudInitNoCloudSource, error) {
+	if callbacks, found := m.callbacksPerHookPoint[hooksInfo.PreCloudInitIsoHookPointName]; found {
+		for _, callback := range callbacks {
+			if callback.Version == hooksV1alpha2.Version {
```

Comment:
The function above `OnDefineDomain` has a a similar if-statment. But wouldn't this cause a panic if a sidecar is on version V1alpha2 and implements `OnDefineDomain`? Because we are on hooksV1alpha2 and the if-statment in `OnDefineDomain` says: `if callback.Version == hooksV1alpha1.Version`.

Reply:
I hadn't thought of this. I had added v1alpha2 support to OnDefineDomain but it was ugly code duplication because I couldn't find a good way to instantiate the client version base on the version var. Since I'm not changing OnDefineDomain I guess I could test if callback.Version is v1alpha1 or v1alpha2 and still use the v1 client. Thoughts?